src/main/java/net/dries007/tfc/world/classic/worldgen/vein/VeinRegistry.java

```diff
@@ -61,53 +61,73 @@ public void preInit(File dir)
             throw new Error("Problem creating TFC extra config directory.");
         }
-        worldGenFile = new File(tfcDir, "ore_spawn_data.json");
         try
         {
-            if (worldGenFile.createNewFile())
-            {
-                FileUtils.copyInputStreamToFile(Objects.requireNonNull(VeinRegistry.class.getClassLoader().getResourceAsStream(DEFAULT_ORE_SPAWN_LOCATION)), worldGenFile);
-            }
+            oreFiles = Files.list(Paths.get(tfcDir.toURI()))
+                .filter(Files::isRegularFile)
+                .map(Path::toFile)
+                .collect(Collectors.toList());
         }
         catch (IOException e)
         {
-            throw new Error("Problem creating default ore vein config file.", e);
+            throw new Error("Problem reading ore vein config files.", e);
+        }
+        if (oreFiles.isEmpty())
+        {
+            // Create default
+            File defaultFile = new File(tfcDir, "ore_spawn_data.json");
+            try
+            {
+                if (defaultFile.createNewFile())
+                {
+                    FileUtils.copyInputStreamToFile(Objects.requireNonNull(VeinRegistry.class.getClassLoader().getResourceAsStream(DEFAULT_ORE_SPAWN_LOCATION)), defaultFile);
+                }
+                oreFiles.add(defaultFile);
+            }
+            catch (IOException e)
+            {
+                throw new Error("Problem creating default ore vein config file.", e);
+            }
         }
     }
 
     public void postInit()
     {
-        String worldGenData;
-        try
-        {
-            worldGenData = FileUtils.readFileToString(worldGenFile, Charset.defaultCharset());
-        }
-        catch (IOException e)
-        {
-            throw new Error("Error reading ore vein config file.", e);
-        }
+        List<String> worldGenData = new ArrayList<>();
+        oreFiles.forEach(file -> {
+            try
+            {
+                worldGenData.add(FileUtils.readFileToString(file, Charset.defaultCharset()));
+            }
+            catch (IOException e)
+            {
+                throw new Error("Error reading " + file, e);
+            }
+        });
 
-        if (Strings.isNullOrEmpty(worldGenData))
-        {
-            TerraFirmaCraft.getLog().warn("The ore vein file is empty! TFC will not generate any ores!");
-        }
-        else
+        if (!worldGenData.isEmpty())
         {
             try
             {
                 weightedVeinTypes.clear();
                 veinTypeRegistry.clear();
-                Map<String, VeinType> values = GSON.fromJson(worldGenData, new TypeToken<Map<String, VeinType>>() {}.getType());
-                values.forEach((name, veinType) -> {
-                    veinType.setRegistryName(name);
-                    veinTypeRegistry.put(name, veinType);
-                    weightedVeinTypes.add(veinType.weight, veinType);
+                worldGenData.forEach(data -> {
```

Comment:
The try / catch block here should be nested as close to the GSON call as possible, so if one file has a syntax error, it doesn't block all files from processing.